### PR TITLE
test: E2E test setup script creates iOS simulators

### PR DESCRIPTION
**Pull request by dcalhoun**
## What?

Expand the E2E test setup script to ensure the required iOS simulators are
available.

## Why?

Relates to https://github.com/WordPress/gutenberg/issues/55251. Simplify setting
up the E2E test environment by automating the process for creating the required
iOS simulators.

## How?

- Convert spaces to tabs to match project
- E2E test script detects or creates iOS simulators
- E2E setup script throws error if required iOS runtime unavailable
- Update E2E test environment setup documentation

## Testing Instructions

**With required simulators unavailable**
1. Delete the iPhone 14 and iPad (10th generation) simulators via `xcrun simctl delete <device-id>`. 
2. Run `npm run native test:e2e:setup`
3. Verify the correct iOS simulators are created. 

**With required simulators available**
1. Verity the iPhone 14 and iPad (10th generation) simulators are available via `xcrun simctl list devices`. 
2. Run `npm run native test:e2e:setup`
3. Verify duplicative simulators are not created. 

### Testing Instructions for Keyboard

n/a

## Screenshots or screencast 
n/a

### CODE REVIEW

File: packages/react-native-editor/bin/test-e2e-setup.sh
Comment (dcalhoun) [inline packages/react-native-editor/bin/test-e2e-setup.sh:1]: I consider myself a shell script novice. Please provide any feedback you may have. 🙇🏻

**Review by geriux — COMMENTED** [review]
Wow this is awesome 👏 Loving the different outputs and its levels (info/error/success) ⭐ 

- ✅  I tested creating the devices and it worked as expected
- ✅  I tested running the script with the devices already created and it worked as expected.
- ⚠️ Tried to run the script after updating the `platformVersion` to a different runtime that I don't have installed and it didn't return the [error in this check](https://github.com/WordPress/gutenberg/pull/55716/files#diff-5fb1a9937f08b0ecd6a7a847922dfb8ff40e5965762a1e59f98ddeaffb99330bR44). It continued to try to create the simulator.

I ran separately:
```
xcrun simctl list runtimes -j | jq -r --arg version "16.9" '.runtimes | to_entries[] | select(.value.version | contains($version))'
```

And it returns an empty string so maybe that's why it skipped the `if` check?

@dcalhoun if you run that command what does it return? Maybe we need to also check that the command's output is not empty 🤔

**Comment by dcalhoun** [comment]
> * ⚠️ Tried to run the script after updating the `platformVersion` to a different runtime that I don't have installed and it didn't return the [error in this check](https://github.com/WordPress/gutenberg/pull/55716/files#diff-5fb1a9937f08b0ecd6a7a847922dfb8ff40e5965762a1e59f98ddeaffb99330bR44). It continued to try to create the simulator.

Good catch! Thank you for testing so thoroughly. I pushed a change that I believe resolves this issue. Will you please review?

**Review by geriux — APPROVED** [review]
This is a major improvement 🤩 LGTM! 🚀

**Pull request merged**